src/main/java/com/greenfoxacademy/service/UserService.java

```diff
@@ -71,10 +73,10 @@ public boolean userLoginIsValid(AuthRequest request) {
     }
 
     public boolean passwordAndEmailMatch(AuthRequest request) {
-        return findUserByEmail(request
-                .getEmail())
-                .getPassword()
-                .equals(encryptPassword(request.getPassword())); // here you should hash the received pw
+        PasswordEncoder passwordEncoder = new BCryptPasswordEncoder();
```

Comment:
@viktornagy create a `@Bean PasswordEncoder` and use `@Autowired PasswordEncoder` here instead